BootloaderCommonPkg/Library/UsbInitLib/UsbInitLib.c

```diff
@@ -109,9 +109,12 @@ InitUsbDevices (
         (Class[2] == PCI_CLASS_SERIAL))) {
       // Enable XHCI controller
       MmioOr8 (PcieAddress + PCI_COMMAND_OFFSET, EFI_PCI_COMMAND_MEMORY_SPACE | EFI_PCI_COMMAND_BUS_MASTER);
-      XhciMmioBase = (UINTN) MmioRead64 (PcieAddress + PCI_BASE_ADDRESSREG_OFFSET) & ~0xF;
 
-      Status = UsbInitCtrl (XhciMmioBase, &mUsbInit.UsbHostHandle);
+      // Read low and high BAR separately
+      XhciMmioBase  = (UINT32)MmioRead32 (PcieAddress + PCI_BASE_ADDRESSREG_OFFSET) & (UINT32)(~0xF);
+      XhciMmioBase |= LShiftU64 ((UINT64)MmioRead32 (PcieAddress + PCI_BASE_ADDRESSREG_OFFSET + 0x4), 32);

```

Comment:
I think your code change is still valid, just need to remove IS_X64 from the condition. 
We still need to check the type type before reading the higher 32 bit.  Only 64 bit bar needs to read higher 32 bit.

Reply:
Added 64-bit BAR type check.